genie/example_filetype_format.py

```diff
@@ -258,7 +258,7 @@ def validate(self, filePathList, **kwargs) -> ValidationResults:
             logger.info("VALIDATING %s" % os.path.basename(",".join(filePathList)))
             errors, warnings = self._validate(df, **mykwargs)
             # only cross-validate if validation passes
-            if not errors:
+            if not errors and self.ancillary_files is not None:
```

Comment:
This will also be an issue if ancillary files is an empty list.  Maybe... this should just be `and self.ancillary_files`

Reply:
@rxu17 does the thumbs up mean you think I should change this line?

Reply:
Sorry, I read your comment incorrectly. Yes, I think `not self.ancillary_files` would be good here then

Reply:
@thomasyu888
Sorry I read your comment incorrectly.
So `not self.ancillary_files` would handle the scenarios where `ancillary_files = None, []`

But we would still need `ancillary_files` to be a list of lists. 
Would something like this help with that:

```suggestion
           is_list_of_lists = all(isinstance(entities, list) for entities in ancillary_files)
           if not errors and isinstance(ancillary_files, list):
               if is_list_of_lists:
                    do stuff
```

Reply:
Hmm, but would this pass?

```
[[]]
```

Reply:
Ah, yeah sorry `parse_file_info_in_nested_list` expects it to not have any empty list in the `ancillary_files` so even [[]], [[],[],[]] or [[], ["something"], []], wouldn't work

Could add a `all(entities for entities in ancillary_files)` to the `is_list_of_lists` check above?

Reply:
@rxu17 Thanks for all the feedback.  After thinking about this some more, I think it makes most sense to refactor to this to use pydantic validators, or have a method to validate the class attributes. 

I am adding mypy static type checking in another branch which will help us catch some of these errors as well, but I'm opting to not over complicate this and pull this in as it stands.